cluster-autoscaler/core/scaleup/equivalence/groups.go

```diff
@@ -76,7 +74,7 @@ func groupPodsBySchedulingProperties(pods []*apiv1.Pod) map[equivalenceGroupId][
 			podEquivalenceGroups[*gid] = append(podEquivalenceGroups[*gid], pod)
 			continue
 		}
-		if len(egs) < maxEquivalenceGroupsByController {
+		if len(egs) < maxPodsPerGroup {
```

Comment:
Wait, this used to be # of equivalence groups per controller and now you're suddenly using max pods per group here. What problem are you trying to solve?